qemu/exec.c

```diff
@@ -1484,14 +1489,17 @@ static void tcg_commit(MemoryListener *listener)
 
 void address_space_init_dispatch(AddressSpace *as)
 {
-    as->dispatch = NULL;
-    as->dispatch_listener = (MemoryListener) {
-        .begin = mem_begin,
-        .commit = mem_commit,
-        .region_add = mem_add,
-        .region_nop = mem_add,
-        .priority = 0,
+    MemoryListener ml = {
+        mem_begin,
+        mem_commit,
+        mem_add,
+        NULL,
+        mem_add,
+        NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+        0,
     };
+    as->dispatch = NULL;
+    as->dispatch_listener = ml;
     memory_listener_register(as->uc, &as->dispatch_listener, as);
```

Comment:
this part (and many other places) can be rewritten like below to (1) keep the change minimal, so minimize mistake. (2) better compatible with future change on the struct of MemoryListener, thus make it easier to port to future Qemu version:

```
MemoryListener ml;

ml.begin = mem_begin;
ml.commit = mem_commit;
ml.region_add = mem_add;
ml.region_nop = mem_add;
ml.priority = 0;

as->dispatch = NULL;
as->dispatch_listener = ml;
```